packages/xrpl/src/models/transactions/transaction.ts

```diff
@@ -395,8 +398,9 @@ export function validate(transaction: Record<string, unknown>): void {
       break
 
     default:
-      throw new ValidationError(
-        `Invalid field TransactionType: ${tx.TransactionType}`,
-      )
+      if (!customDefinition)
+        throw new ValidationError(
```

Comment:
Can this check the list of transactions in `customDefinition` instead of never erroring?

Reply:
Or maybe `validate` should be skipped entirely in the custom definition case?

Reply:
> Can this check the list of transactions in `customDefinition` instead of never erroring?

I like this idea. I will add it.

Reply:
> > Or maybe `validate` should be skipped entirely in the custom definition case?
> 
> I thought about it, but then I thought there were other checks that might be useful.

True, but there may be new transactions that break those checks.

Just a thought - I don't have a strong preference on this.